pumpkin/src/world/mod.rs

```diff
@@ -349,4 +355,39 @@ impl World {
         self.broadcast_packet_all(&CRemoveEntities::new(&[entity.entity_id.into()]))
             .await
     }
+    pub async fn set_block(&self, position: WorldPosition, block_id: BlockId) {
+        let (z_chunk, z_rem) = position.0.z.div_rem_euclid(&16);
+        let (x_chunk, x_rem) = position.0.x.div_rem_euclid(&16);
+        let chunk_coordinate = Vector2 {
+            x: x_chunk,
+            z: z_chunk,
+        };
```

Comment:
I would like to see stuff like this turned into helper functions on `WorldPosition`. 

It can be a `position.get_chunk_position()` (naming here might be confusing for in-chunk position vs chunk position) or similar.

Reply:
I also thought of this, but it must've slipped my mind.